lib/RelatedResourceProviders/FilesRelatedResourceProvider.php

```diff
@@ -233,4 +200,36 @@ private function getItemIdsFromParentPath(int $itemId): array {
 
 		return $itemIds;
 	}
+
+
+	/**
+	 * @param RelatedResource $related
+	 * @param FilesShare $share
+	 */
+	private function processShareRecipient(RelatedResource $related, FilesShare $share) {
+		try {
+			$sharedWith = $this->miscService->convertShareRecipient(
+				$share->getShareType(),
+				$share->getSharedWith()
+			);
+
+			if ($share->getShareType() === IShare::TYPE_USER) {
+				$shareCreator = $this->miscService->convertShareRecipient(
+					IShare::TYPE_USER,
+					$share->getShareCreator()
+				);
```

Comment:
Why is creator added to virtual group for files but not deck or calendar?

Reply:
The owner of the share is not defined the same way for deck/calendar, so I am using the owner of the entry: https://github.com/nextcloud/related_resources/blob/fc1bf6216003cc2374d8d5d95a77153336eb7510/lib/RelatedResourceProviders/DeckRelatedResourceProvider.php#L99